plugins/module_utils/core.py

```diff
@@ -357,3 +357,27 @@ def get_boto3_client_method_parameters(client, method_name, required=False):
     else:
         parameters = list(input_shape.members.keys())
     return parameters
+
+
+def scrub_none_parameters(parameters):
+    """
+    Iterate over a dictionary removing any keys that have a None value
+
+    Reference: https://github.com/ansible-collections/community.aws/issues/251
+    Credit: https://medium.com/better-programming/how-to-remove-null-none-values-from-a-dictionary-in-python-1bedf1aab5e4
+
+    :param parameters: parameter dict
+    :return: parameter dict with all keys = None removed
+    """
+
+    clean_parameters = {}
+
+    for k, v in parameters.items():
+        if isinstance(v, dict):
+            nested = scrub_none_parameters(v)
+            if len(nested.keys()) > 0:
```

Comment:
The more Pythonic way would be:
```suggestion
            if nested:
```
In any case: is this needed? I don't know the AWS APIs, but maybe there are cases where an empty dictionary is treated differently from a not specified one.

Reply:
Even if not in the AWS APIs I know that modules themselves depend on this behaviour.  For example, an empty dict may remove all tags where as an unspecified dict will leave tags alone.